core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -552,7 +552,9 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
       taskId: String,
       reason: String): Unit = {
     stateLock.synchronized {
-      removeExecutor(taskId, SlaveLost(reason))
+      if (!stopCalled) {
```

Comment:
OK you've convinced me.  But please add a clarifying comment to super.removeExecutor() specifying that it should not becalled after super.stop() is called.